### PR TITLE
In Stripe module, skip donation code when disabled

### PR DESCRIPTION
**Pull request by jmoldow**
Previously, if the Stripe module setting ``offer_donation`` was set to ``False``, only the accounting controller and template would be affected.

Now, the donation setup that happens in the module will also be skipped when ``offer_donation`` is ``False``.

This change has been used successfully for multiple Yale programs.

### CODE REVIEW

**Comment by benjaminjkraft** [comment]
Just to check, this will all work right if a chapter enables the module without donations, and later in the program turns on donations, right?

File: esp/esp/program/modules/handlers/creditcardmodule_stripe.py [hunk 1]
```diff
@@ -158,8 +159,9 @@ def payonline(self, request, tl, one, two, module, extra, prog):
         payment_type = iac.default_payments_lineitemtype()
         sibling_type = iac.default_siblingdiscount_lineitemtype()
         grant_type = iac.default_finaid_lineitemtype()
-        donate_type = iac.get_lineitemtypes().get(text=self.settings['donation_text'])
-        context['itemizedcosts'] = iac.get_transfers().exclude(line_item__in=[payment_type, sibling_type, grant_type, donate_type]).order_by('-line_item__required')
+        offer_donation = self.settings['offer_donation']
+        donate_type = iac.get_lineitemtypes().get(text=self.settings['donation_text']) if offer_donation else None
+        context['itemizedcosts'] = iac.get_transfers().exclude(line_item__in=filter(None, [payment_type, sibling_type, grant_type, donate_type])).order_by('-line_item__required')
```
Comment (benjaminjkraft) [inline esp/esp/program/modules/handlers/creditcardmodule_stripe.py]: > `filter(None, ...)`

Huh, TIL.

**Comment by jmoldow** [comment]
> Just to check, this will all work right if a chapter enables the module without donations, and later in the program turns on donations, right?

It should. The ``check_setup()`` function gets run on every request. This will create the needed ``LineItemType`` before anything else happens.

**Comment by benjaminjkraft** [comment]
Cool, LGTM.

**Pull request merged**